src/h3lib/lib/polygon.c

```diff
@@ -26,111 +26,86 @@
-                       const GeoCoord* coord) {
-    // fail fast if we're outside the bounding box
-    if (!bboxContains(bbox, coord)) {
-        return false;
-    }
-    bool isTransmeridian = bboxIsTransmeridian(bbox);
-    bool contains = false;
-
-    double lat = coord->lat;
-    double lng = _normalizeLng(coord->lon, isTransmeridian);
-
-    for (int i = 0; i < geofence->numVerts; i++) {
-        GeoCoord a = geofence->verts[i];
-        GeoCoord b;
-        if (i + 1 == geofence->numVerts) {
-            b = geofence->verts[0];
-        } else {
-            b = geofence->verts[i + 1];
-        }
 
-        // Ray casting algo requires the second point to always be higher
-        // than the first, so swap if needed
-        if (a.lat > b.lat) {
-            a = b;
-            b = geofence->verts[i];
-        }
+/**
+ * Take a given LinkedGeoLoop data structure and check if it
+ * contains a given geo coordinate.
+ * @name pointInsideLinkedGeoLoop
+ *
+ * @param loop          The linked loop
+ * @param bbox          The bbox for the loop
+ * @param coord         The coordinate to check
+ * @return              Whether the point is contained
+ */
 
-        // If we're totally above or below the latitude ranges, the test
-        // ray cannot intersect the line segment, so let's move on
-        if (lat < a.lat || lat > b.lat) {
-            continue;
-        }
+/**
+ * Create a bounding box from a LinkedGeoLoop
+ * @name bboxFromLinkedGeoLoop
+ * @param geofence Input Geofence
+ * @param bbox     Output bbox
+ */
 
-        double aLng = _normalizeLng(a.lon, isTransmeridian);
-        double bLng = _normalizeLng(b.lon, isTransmeridian);
+// Define macros used in polygon algos for Geofence
+#define TYPE Geofence
+#define INIT_ITERATION INIT_ITERATION_GEOFENCE
+#define ITERATE ITERATE_GEOFENCE
+#define IS_EMPTY IS_EMPTY_GEOFENCE
 
-        // Rays are cast in the longitudinal direction, in case a point
-        // exactly matches, to decide tiebreakers, bias westerly
-        if (aLng == lng || bLng == lng) {
-            lng -= DBL_EPSILON;
-        }
+#include "polygonAlgos.h"
 
-        // For the latitude of the point, compute the longitude of the
-        // point that lies on the line segment defined by a and b
-        // This is done by computing the percent above a the lat is,
-        // and traversing the same percent in the longitudinal direction
-        // of a to b
-        double ratio = (lat - a.lat) / (b.lat - a.lat);
-        double testLng =
-            _normalizeLng(aLng + (bLng - aLng) * ratio, isTransmeridian);
-
-        // Intersection of the ray
-        if (testLng > lng) {
-            contains = !contains;
-        }
+/**
+ * Create a bounding box from a GeoPolygon
+ * @param polygon Input GeoPolygon
+ * @param bboxes  Output bboxes, one for the outer loop and one for each hole
+ */
+void bboxesFromGeoPolygon(const GeoPolygon* polygon, BBox* bboxes) {
```

Comment:
Why are these functions not generated?

Reply:
Because the generated functions deal with loops (either `Geofence` or `LinkedGeoLoop` structs), which are children of polygons. None of the polygon functions are or need to be polymorphic.